Remove WKWebsitePoliciesSetCustomHeaderFields

achristensen07 opened the pull request:
#### 21027fde1383bfb7471742e6a23d28ceaad4ca91
<pre>
Remove WKWebsitePoliciesSetCustomHeaderFields
<a href="https://bugs.webkit.org/show_bug.cgi?id=243698">https://bugs.webkit.org/show_bug.cgi?id=243698</a>

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
(API::WebsitePolicies::data):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/C/WKWebsitePolicies.cpp:
(WKWebsitePoliciesCopyCustomHeaderFields):
(WKWebsitePoliciesSetCustomHeaderFields):
* Source/WebKit/UIProcess/API/C/WKWebsitePolicies.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm:
(-[CustomHeaderFieldsDelegate webView:decidePolicyForNavigationAction:preferences:decisionHandler:]):
(-[CustomHeaderFieldsDelegate webView:startURLSchemeTask:]):
(expectLegacyHeaders): Deleted.

Canonical link: <a href="https://commits.webkit.org/253245@main">https://commits.webkit.org/253245@main</a>
</pre>

hortont424 commented on Source/WebKit/UIProcess/API/C/WKWebsitePolicies.cpp:
```diff
@@ -56,27 +56,13 @@ bool WKWebsitePoliciesGetContentBlockersEnabled(WKWebsitePoliciesRef websitePoli
-WK_EXPORT WKDictionaryRef WKWebsitePoliciesCopyCustomHeaderFields(WKWebsitePoliciesRef websitePolicies)
+WK_EXPORT WKDictionaryRef WKWebsitePoliciesCopyCustomHeaderFields(WKWebsitePoliciesRef)
 {
-    HashMap<WTF::String, RefPtr<API::Object>> fields;
```
Title should clarify that you are not just deprecating but actually de-implementing.

achristensen07 committed:
Remove WKWebsitePoliciesSetCustomHeaderFields
https://bugs.webkit.org/show_bug.cgi?id=243698

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
(API::WebsitePolicies::data):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/C/WKWebsitePolicies.cpp:
(WKWebsitePoliciesCopyCustomHeaderFields):
(WKWebsitePoliciesSetCustomHeaderFields):
* Source/WebKit/UIProcess/API/C/WKWebsitePolicies.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm:
(-[CustomHeaderFieldsDelegate webView:decidePolicyForNavigationAction:preferences:decisionHandler:]):
(-[CustomHeaderFieldsDelegate webView:startURLSchemeTask:]):
(expectLegacyHeaders): Deleted.

Canonical link: https://commits.webkit.org/253245@main
